// File:        conf_deploy.proto
// Description: ---
// Notes:       ---
// Author:      leoxiang <leoxiang727@qq.com>
// Revision:    2015-06-09 by leoxiang

package conf;

////////////////////////////////////////////////////////////
// Enum define
enum ENServerType
{
    SERVER_TYPE_CONFIG_CENTER = 1;
    SERVER_TYPE_PROXY_SERVER = 2;
    SERVER_TYPE_LOG_SERVER = 3;
    SERVER_TYPE_CONN_SERVER = 4;
    SERVER_TYPE_GAME_SERVER = 5;
    SERVER_TYPE_ROOM_SERVER = 6;
    SERVER_TYPE_DB_SERVER = 7;
}

enum ENServiceType
{
    SERVICE_TYPE_CLIENT_MSG     = 9; // receive client msg and response
    SERVICE_TYPE_CONN_ACTIVE    = 1; // transfer client message
    SERVICE_TYPE_CONN_PASSIVE   = 2; // transfer client message
    SERVICE_TYPE_PROXY_ACTIVE   = 3; // transfer server to server message
    SERVICE_TYPE_PROXY_PASSIVE  = 4; // transfer server to server message
    SERVICE_TYPE_LOG_ACTIVE     = 5; // process net log
    SERVICE_TYPE_LOG_PASSIVE    = 6; // process net log
    SERVICE_TYPE_CONFIG_ACTIVE  = 7; // request and push configs
    SERVICE_TYPE_CONFIG_PASSIVE = 8; // request and push configs
}

message Service
{
    optional ENServiceType service_type = 1;
    repeated string listen_address = 2;
    repeated string connect_address = 3;
}

message Server
{
    optional ENServerType server_type = 1;
    optional uint32       server_id = 2; 
    optional string       local_log_path = 3;
    repeated Service      services = 4;
}

message Deploy
{
    optional Server default_config = 1;
    repeated Server server_config = 2;
}

// vim:ts=4:sw=4:et:ft=proto:
